### PR TITLE
Convert create-tigris-app to TypeScript and add support for templates

### PR DESCRIPTION
**Pull request by ovaistariq**
ovaistariq@Ovaiss-MBP::~/code/tigrisdata/create-tigris-app (updates)
$ npm run build

> @tigrisdata/create-tigris-app@1.0.0-beta.9 prebuild
> rimraf ./dist/

> @tigrisdata/create-tigris-app@1.0.0-beta.9 build
> npx tsc && chmod +x ./dist/index.js

ovaistariq@Ovaiss-MBP::~/code/tigrisdata/create-tigris-app (updates)
$ ./dist/index.js
✔ What is your project named? … /Users/ovaistariq/projects/my-app4
✔ What is the clientId? … abc
✔ What is the clientSecret? … ***
Creating a new app in /Users/ovaistariq/projects/my-app4.

Using npm.

Installing dependencies:
- @tigrisdata/core
- dotenv
- typescript
- ts-node
- eslint

added 149 packages, and audited 150 packages in 6s

25 packages are looking for funding
  run `npm fund` for details

found 0 vulnerabilities

Initializing project with template: default

Initialized a git repository.

Success! Created my-app4 at /Users/ovaistariq/projects/my-app4

### CODE REVIEW

**Review by tahakhan — APPROVED** [review]
Tested and works great!

File: templates/default/setup.ts [hunk 1]
```diff
@@ -0,0 +1,21 @@
+import * as dotenv from "dotenv";
+dotenv.config();
```
Comment (adilansari) [inline templates/default/setup.ts:2]: Tigris SDK will automatically do this for you. Is this for the use case when Tigris SDK package is in a different path in local filesystem?

File: helpers/git.ts [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+      return false;
+    }
+
+    execSync("git init", { stdio: "ignore" });
+    didInit = true;
+
+    execSync("git checkout -b main", { stdio: "ignore" });
+
+    execSync("git add -A", { stdio: "ignore" });
+    execSync('git commit -m "Initial commit from Create Next App"', {
```
Comment (adilansari) [inline helpers/git.ts:36]: Create Tigris App

**Review by adilansari — APPROVED** [review]

File: helpers/git.ts (hunk 2 above)
Comment (ovaistariq) [reply helpers/git.ts]: FIxed

File: templates/default/setup.ts (hunk 1 above)
Comment (ovaistariq) [reply templates/default/setup.ts]: Fixed

**Pull request merged**